mirrord-layer/src/socket/ops.rs

```diff
@@ -0,0 +1,210 @@
+use std::{
+    os::unix::{io::RawFd, prelude::*},
+    sync::Arc,
+};
+
+use libc::{c_int, sockaddr, socklen_t};
+use os_socketaddr::OsSocketAddr;
+use socket2::{Domain, Protocol, SockAddr, Socket, Type};
+use tracing::{debug, error, trace, warn};
+
+use super::*;
+use crate::{
+    blocking_send_hook_message,
+    error::LayerError,
+    message::HookMessage,
+    tcp::{HookMessageTcp, Listen},
+    HOOK_SENDER,
+};
+
+/// Create the socket, add it to `MIRROR_SOCKETS` if it's successful, plus protocol and domain
+/// matches our expected type (Tcpv4/v6).
+///
+/// Otherwise the socket is added to `BYPASS_SOCKETS`.
```

Comment:
why do we need `BYPASS_SOCKETS`?

Reply:
Good intuition, we don't need it. I started up with the assumption that we would need to capture every socket, but I don't think we need to do so.